refactor(ngSanitize) Remove workarounds for IE8

realityking opened the pull request:
Also provides better tests.

realityking committed:
refactor(ngSanitize): Remove workarounds for IE8

Narretz commented:
What do you mean by "better tests"? I assume you removed those tests that were testing the IE8 specific functions?

realityking commented:
@Narretz Yes, I removed those testing IE8 specific workarounds and added some tests to tests the general functionality - otherwise there would have been no tests left.

Narretz commented:
Okay, then you should split this into commits - the first includes the new tests, the second removes all IE8 stuff.

realityking committed:
test(ngSanitize): Add tests for decodeEntities

realityking commented:
@Narretz done

realityking commented:
@Narretz The only failed job seems to be related to saucelabs. Can this be merged?

Narretz commented:
Hi, sorry for the late response. I have another very small nitpick: our commit messages have a `:` after the `refactor(ngSanitize)` part, can you fix that up?

@petebacondarwin are you okay with removing this IE8 workaround?

realityking commented:
@Narretz Heh, never noticed that. Fixed now.

petebacondarwin commented:
@realityking - more commit message nitpicking :-) We don't capitalize the first letter of the description.

That being said, I am happy to help strip out unnecessary code from the codebase and get this in.
Thanks!

petebacondarwin commented on test/ngSanitize/sanitizeSpec.js:
```diff
@@ -511,37 +510,13 @@ describe('decodeEntities', function() {
-  afterEach(function() {
-    window.hiddenPre = origHiddenPre;
+  it('should unescape text', function() {
+    htmlParser('a&lt;div&gt;&amp;&lt;/div&gt;c', handler)
```
missing semi-colon

petebacondarwin commented on test/ngSanitize/sanitizeSpec.js:
```diff
@@ -511,37 +510,13 @@ describe('decodeEntities', function() {
-      expect(text).toEqual('');
-    });
+  it('should preserve whitespace', function() {
+    htmlParser('  a&amp;b ', handler)
```
missing semi-colon

petebacondarwin commented:
It is always good to run `grunt test` on our PRs before submitting. Apart from building, minifying and running all the unit and e2e tests locally, it also runs `grunt ci-checks` which checks the jshint and jscs rules.

realityking commented:
@petebacondarwin I always run `grunt test` but I hadn't yet discovered `grunt ci-checks`, thanks for the pointer.
